### PR TITLE
Use o-layout, update sidebar content.

### PR DESCRIPTION
**Pull request by notlee**
**Changes**
- Uses `o-layout`.
- Updates sidebar content.
- Upgrades to the latest version of `o-message`

Design and content changes are driven by our user research from earlier in the year, and the constrained width of `o-layout` -- for a more seamless experience when browsing Origami documentation.

**Next Up**
- This PR became huge, so I've resisted upgrading `o-labels` and `o-header-services` for a future PR.

**Component Overview Before/After**
screenshot 2019-02-12 at 14 40 37
screenshot 2019-02-12 at 14 40 40

**Component Demos Before/After**
screenshot 2019-02-12 at 14 41 45
screenshot 2019-02-12 at 14 41 49

**Component Readme Before/After**
screenshot 2019-02-12 at 14 42 29
screenshot 2019-02-12 at 14 42 26

**Component Codedocs Before/After**
screenshot 2019-02-12 at 14 43 15
screenshot 2019-02-12 at 14 43 17

**Imageset Before/After**
screenshot 2019-02-12 at 14 44 03
screenshot 2019-02-12 at 14 44 06

**Service Before/After**
screenshot 2019-02-12 at 14 45 17
screenshot 2019-02-12 at 14 45 23
screenshot 2019-02-12 at 14 45 28

### CODE REVIEW

File: src/main.scss [hunk 1]
```diff
@@ -20,43 +20,104 @@ $o-brand: 'internal';
...
-		border-bottom: 1px solid oColorsGetPaletteColor('slate');
-		text-transform: uppercase;
+.registry-title span {
+	vertical-align: middle;
+}
+
+details {
+	margin-bottom: oTypographySpacingSize(4);
+	summary {
+		@include oTypographyItalic();
```
Comment (JakeChampion) [inline src/main.scss]: I'm of the opinion that we shouldn't use italic, I also don't know why we are using italic.
If we want different styles from the paragraph inside the details element when it is open we can use `details[open] summary { font-weight: something}`. I tried using `text-decoration` but that seems to not work with summary elements inside details elements /shrug.

Comment (notlee) [reply src/main.scss]: I think I prefer italics, I think it works for secondary FAQ content, but don't feel too strongly. I've updated to the font weight on click approach...

Before:
italics

After:
bold

**Comment by notlee** [comment]
I suggest clicking around to review the experience as a whole. There are a handful of small chnages you'll come across, such as updated messaging when a component does not support the default brand:

screenshot 2019-02-12 at 14 53 54

File: lib/code-docs/jsdoc/nav.js [hunk 2]
```diff
@@ -23,10 +24,13 @@ class JsDocNav {
         docletGroups.forEach(docletGroup => {
             if (jsDocByTypeWithMembers[docletGroup.name]) {
                 jsDocByTypeWithMembers[docletGroup.name].forEach((node) => {
-                    nav.push(createSubNav(
+                    const navnode = new NavNode(node.name);
+                    const subnav = createSubNav(
                         `${node.name}`,
                         [].concat([node], ...docletGroup.subGroups.map(name => node[name]))
```
Comment (JakeChampion) [inline lib/code-docs/jsdoc/nav.js]:
```suggestion
                        [].concat([node], docletGroup.subGroups.map(name => node[name]))
```

Comment (JakeChampion) [inline lib/code-docs/jsdoc/nav.js]:
```suggestion
                        [node].concat(docletGroup.subGroups.map(name => node[name]))
```

**Comment by rowanmanning** [comment]
Some things we discussed in person:

1. Could we introduce the GitHub logo to the GitHub link? There's some disagreement on the prominence of this and I wondered whether a logo might be a good compromise
2. Sidebar items can overflow the sidebar (see o-typography SassDoc)
3. Sidebar items can appear really small (genuinely not sure if we can fix both this and the above 🤷‍♂️)
4. Use o-header-services v3, and make the nav consistent with the new website, also both places need the Origami logo rather than FT

**Comment by notlee** [comment]
- I have a PRs open for point 1 and 4. 
- Point 3 I don't think we can change either, if we made the width conditional on content the sidebar would jump around whilst navigating the site. 
- Point 4 actually seems to be okay (were we previewing an old demo?) It's not super ideal but long nav items wrap rather than overflow ([see here](https://origami-registry-ui-qa-pr-144.herokuapp.com/components/o-typography@5.10.1/sassdoc?brand=master#mixin-otypographyprogressivefontfallback)). 
screenshot 2019-03-04 at 10 29 01

**Pull request merged**